File: book_this-is-coding-test-with-python/part2/chap4-implementation/ex1/answer.py
n = int(input())
commands = input().split()

rx = ry = 1


def left():
    global rx
    if rx == 1:
        return
    rx -= 1


def right():
    global rx
    if rx == n:
        return
    rx += 1


def up():
    global ry
    if ry == 1:
        return
    ry -= 1


def down():
    global ry
    if ry == n:
        return
    ry += 1


for command in commands:
    if command == "L":
        left()
    elif command == "R":
        right()
    elif command == "U":
        up()
    elif command == "D":
        down()

print(rx, ry)

File: book_this-is-coding-test-with-python/part2/chap4-implementation/ex1/answer.js
let coords = {
  x: 1,
  y: 1,

  left: function () {
    if (this.x !== 1) this.x -= 1;
  },
  right: function () {
    if (this.x !== this.maxX) this.x += 1;
  },
  up: function () {
    if (this.y !== 1) this.y -= 1;
  },
  down: function () {
    if (this.y !== this.maxY) this.y += 1;
  },
};

function solution(n, commands) {
  coords.maxX = n;
  coords.maxY = n;

  for (const command of commands) {
    if (command === "L") coords.left();
    else if (command === "R") coords.right();
    else if (command === "U") coords.up();
    else if (command === "D") coords.down();
  }

  return { x: coords.x, y: coords.y };
}

module.exports = solution;
